src/hooks/useNotificationSound.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react";
+import useSound from "use-sound";
+
+import notificationSound from "assets/sounds/newmessage.m4a";
+
+import { PrivateChatMessage, ChatTypes } from "types/chat";
+
+import { useChatSidebarControls } from "hooks/chatSidebar";
+
+import { useUser } from "./useUser";
+import { WithId } from "utils/id";
+
+export const useNotificationSound = (
+  privateChatMessages?: WithId<PrivateChatMessage>[]
+) => {
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(notificationSound, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const [privateChatMessagesState, setPrivateChatMessagesState] = useState(
+    privateChatMessages
+  );
+
+  const chatWithUser =
+    chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT
+      ? chatSettings.recipientId
+      : undefined;
```

Comment:
Maybe rename this something more informative such as `currentChatUserId` or similar